ibtk/include/ibtk/LibMeshSystemIBVectors.h

```diff
@@ -0,0 +1,129 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2020 - 2020 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+#ifndef included_IBTK_LibMeshSystemIBVectors
+#define included_IBTK_LibMeshSystemIBVectors
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibtk/FEDataManager.h>
+#include <ibtk/LibMeshSystemVectors.h>
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBTK
+{
+/*!
+ * \brief Class LibMeshSystemIBVectors is a convenience class that manages
+ * access to libMesh vectors for the same system defined on multiple
+ * parts. This class supports access to vectors ghosted with the Lagrangian
+ * partitioning (i.e., libMesh's computed partitioning) as well as the IB
+ * partitioning (i.e., the partitioning based on the distribution of SAMRAI
+ * data).
+ *
+ * A libMesh::System stores vectors in two different ways: the
+ * <tt>solution</tt> and <tt>current_local_solution</tt> vectors are stored
+ * explicitly by the System object (see libMesh::System::solution and
+ * libMesh::System::current_local_solution). They are <em>not</em> accessible
+ * by libMesh::System::request_vector() and related functions. All other
+ * vectors (including the right-hand side) are stored inside a map and are
+ * accessible by <tt>request_vector()</tt>. IBTK::LibMeshSystemIBVectors::get(),
+ * for the sake of a uniform interface, is inconsistent with libMesh in this
+ * regard: the vector name <tt>"solution"</tt> refers to the solution vector
+ * (which is not stored in the map). The same is true for
+ * <tt>"current_local_solution"</tt>. For convenience, this class also accepts
+ * <tt>"current"</tt> as an alias for <tt>"current_local_solution"</tt>. All
+ * other vector names are stored in the libMesh::System vector map.
+ *
+ * All vectors added to a libMesh::System by this class are ghosted (with
+ * ghost data computed by the System) and registered for projection (i.e., if
+ * the mesh changes, these vectors will be automatically updated).
+ *
+ * This class stores information that depends on SAMRAI's parallel
+ * partitioning: LibMeshSystemIBVectors::reinit() should be called after
+ * regridding to clear this data.
+ *
+ * @note This class is intended for internal use in IBAMR. It is used in
+ * IBFEMethod and IBFESurfaceMethod.
```

Comment:
The class documentation needs to be updated so that it explains what is different in this class vs. the base class.

Reply:
What was changed here?

Reply:
Not much.

Reply:
I think it is still correct though. :-)

Reply:
It is but I would prefer to delete all the repeated text so that this doc string only explains the new stuff.

Reply:
OK I got rid of duplicated text.

Reply:
:heavy_check_mark: